plugins/SegmentEditor/Model.php

```diff
@@ -129,6 +129,74 @@ public function getSegmentByDefinition($definition)
         return $segment;
     }
 
+    /**
+     * Gets a list of segments that have been deleted in the last week and therefore may have orphaned archives.
+     * @param Date $date Segments deleted on or after this date will be returned.
+     * @return array of segments. The segments are only populated with the fields needed for archive invalidation
+     * (e.g. definition, enable_only_idsite).
+     * @throws \Exception
+     */
+    public function getSegmentsDeletedSince(Date $date)
+    {
+        $dateStr = $date->toString('Y-m-d');
+        $sql = "SELECT DISTINCT definition, enable_only_idsite FROM " . Common::prefixTable('segment')
+            . " WHERE deleted = 1 AND ts_last_edit >= '$dateStr'";
```

Comment:
here we want to use bound parameters... meaning `AND ts_last_edit >= ?` and then below we use `Db::fetchAll($sql, array($dateStr))`. In this case it wouldn't be a security issue to not use bound parameters but it's still better in general and makes sure it can't become a security issue in the future